connector/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala

```diff
@@ -40,6 +40,7 @@ import org.apache.spark.internal.Logging
  *
  * PLEASE KEEP THIS FILE UNDER src/main AS PYTHON TESTS NEED ACCESS TO THIS FILE!
  */
+@deprecated("This class will be move to `test`.", "3.5.2")
 private[kinesis] class KinesisTestUtils(streamShardCount: Int = 2) extends Logging {
```

Comment:
are these APIs? They are `private[kinesis]`

Reply:
Oh, it's my bad. You're right. Only `WriteInputFormatTestDataGenerator` is the valid one.

Reply:
To @panbingkun , sorry but could you remove the ones on `private[kinesis]`?

Reply:
> are these APIs? They are `private[kinesis]`

+1

Reply:
Does this class also need to be `marked`?
<img width="495" alt="image" src="https://github.com/apache/spark/assets/15246973/7081e2ea-a696-419d-97ea-8ea6d001456c">

Reply:
Yes, that needs mark.

Reply:
Okay

Reply:
All done.